pkg/drivers/generic/metrics.go

```diff
@@ -0,0 +1,43 @@
+package generic
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	SQLTotal = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "kine_sql_total",
+		Help: "Total number of SQL operations",
+	})
+
+	SQLErrors = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "kine_sql_errors_total",
+		Help: "Total number of SQL operation errors",
+	})
+
+	SQLTime = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Name: "kine_sql_time_seconds",
+		Help: "Length of time per SQL operation",
+		Buckets: []float64{0.005, 0.01, 0.025, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0,
+			1.5, 2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5, 6, 7, 8, 9, 10, 15, 20, 25, 30},
+	})
+)
+
+const (
+	slowSQLThreshold = time.Second
```

Comment:
This should probably be configurable somehow, rather than hardcoded. See https://github.com/k3s-io/kine/blob/dc877c3e2003aa6e01f1e35a09a6fb2578aa8a6e/pkg/drivers/jetstream/jetstream.go#L68